ouroboros-network/src/Ouroboros/Network/Subscription/Common.hs

```diff
@@ -268,17 +304,14 @@ subscriptionWorker tbl tracer localIPv4 localIPv6 connectionAttemptDelay getTarg
   where
     worker :: TVar IO (Set ThreadId) -> IO ()
     worker childrenVar = do
-        valencyVar <- newTVarM valency
+        valencyVar <- atomically $ newValencyCounter tbl valency
```

Comment:
This is the only place where a new valency counter is created.  It seems that the `vcId` field is not really needed, since there is only a one `ValencyCounter` for the entire `subscriptionWorker`.

Reply:
There is only one valency counter for a subscription worker but there can be multiple subscribers per connection table and a valency counter will exists in multiple places in the connection table.
Example Toy config, for 3 workers with two active connections spread over 3 timezones:
Domain: emea-mainnet.iohk.io, port 1234, Valency 2
Domain: na-mainnet.iohk.io, port 1234, Valency 2
Domain: asia-mainnet.iohk.io, port 1234, Valency 2


The `vcId` field is still needed to prevent double. counting of connections as described in the commit  https://github.com/input-output-hk/ouroboros-network/pull/768/commits/481c25741e55b4bc414f52060910aaf1b87fef01